File: koma-cli/src/main/java/org/koma/cli/util/IO.java
package org.koma.cli.util;

import lombok.experimental.UtilityClass;
import lombok.extern.slf4j.Slf4j;
import lombok.val;
import org.apache.commons.io.FileUtils;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.nio.file.Path;

@UtilityClass
@Slf4j
public class IO {

  public void checkDirExistsOrCreate(@NotNull Path dir) {
    val dirFile = dir.toAbsolutePath().toFile();
    if (!dirFile.exists()) {
      if (dirFile.mkdir()) {
        log.info("created {}", dir.toAbsolutePath().toString());
      }
    }
  }
}

File: koma-cli/src/main/kotlin/org/koma/cli/util/IO.kt
package org.koma.cli.util

import io.github.oshai.kotlinlogging.KotlinLogging
import java.nio.file.Path

val log = KotlinLogging.logger {}

fun checkDirExistsOrCreate(dir: Path) {
  val dirFile = dir.toAbsolutePath().toFile()
  if (dirFile.exists()) return
  if (dirFile.mkdir()) {
    log.info { "created ${dir.toAbsolutePath()}" }
  }
}
